docs/simulators/OpenFOAM.md

```diff
@@ -126,6 +126,286 @@ task.download_outputs()
+runApplication rm -r constant/polyMesh.origHalf
+runApplication cp -r 0.orig 0
+runApplication cd system
+runApplication cp system/controlDict.SHM system/controlDict
+runApplication cp system/fvSchemes.SHM system/fvSchemes
+runApplication cp system/fvSolution.SHM system/fvSolution
+runApplication decomposePar
+runParallel renumberMesh -overwrite
+runApplication cp system/controlDict.SRS.init system/controlDict
+runApplication cp system/fvSolution.SRS system/fvSolution
+runApplication cp system/fvSchemes.SRS system/fvSchemes
+runParallel applyBoundaryLayer -ybl 0.1
+runApplication cp system/controlDict.SRS.init system/controlDict
+runApplication cp system/fvSolution.SRS system/fvSolution
+runApplication cp system/fvSchemes.SRS system/fvSchemes
+runParallel rhoPimpleFoam
+runApplication cp system/controlDict.SRS.avg system/controlDict
+runApplication cp system/fvSolution.SRS system/fvSolution
+runApplication cp system/fvSchemes.SRS system/fvSchemes
+runParallel rhoPimpleFoam
+runParallel postProcess -func sampleDict.surface.SRS -latestTime
+```
+
+Below is the Python code to configure and run the OpenFOAM simulation:
+
+```python notest
+import inductiva
+import os
+
+machine_group = inductiva.resources.MachineGroup(machine_type="c3d-highcpu-360")
+machine_group.start()
+
+import inductiva
+
+# Set simulation input directory
+input_dir = "/path/to/highLiftConfiguration"
+
+# Read the simulation commands
+with open(os.path.join(input_dir,'commands.txt'), 'r') as file:
+    commands = [line.strip() for line in file]
+
+# Initialize the Simulator
+openfoam = inductiva.simulators.OpenFOAM(distribution="esi")
+
+# Run simulation
+task = openfoam.run(
+    input_dir=input_dir,
+    commands=commands,
+    n_vcpus=180,
+    use_hwthread=True,
+    on=machine_group)
+
+# Wait for the task to finish and downloading the outputs
+task.wait()
+task.download_outputs()
+
+# Turn off the machine group
+machine_group.terminate()
+
+```
+
+### Important Details
+
+1. **ExaFOAM Benchmark**: This simulation replicates the MB9 micro-benchmark
+from the [ExaFOAM](https://exafoam.eu/benchmarks/) project. The benchmark
+focuses on capturing key flow physics and the simulation approach of a larger
+HPC challenge with fewer computational resources, providing a scalable and
+efficient simulation of a high-lift configuration.
+   
+2. **Machine Configuration**: We are using a powerful `c3d-highcpu-360`
+machine with 360 vCPUs to ensure that the computation proceeds efficiently.
+You can adjust this depending on the available resources or your project's needs.
+   
+3. **Commands**: The list of `commands` involves several key OpenFOAM tasks
+like copying necessary configuration files, running mesh generation, and
+parallelizing the process using `decomposePar` and `snappyHexMesh`.
+
+4. **Parallelization**: We use `n_vcpus=180` in the simulation, which will
+allow OpenFOAM to run in parallel, greatly speeding up the computation.
+
+### Running the Simulation
```

Comment:
@Pbarbosa92 for lists of commands, we would love to know which command is being executed at each time. It could show up on the console or on the CLI. Please own this UX issue and chat with @ClaudiaRSilva about this.

Reply:
We have that on the stdout and stderr.
We have the command and the respective output.

Were do you want this information to show ?

Reply:
It should show as output of the method wait(). It should be interactive.

"Executing command XXXXX".

Ideally we would have that showing on the console as the task updates.

Reply:
I believe this would be doable. A task could have a field called `command` that saves the current command being executed at each time (the executer tracker would need to update this field).

I will go further into this ( i will check with the back end the effort needed to implement this and will create and maybe implement the changes needed)